i8_terminal/commands/earnings/earnings_list.py

```diff
@@ -60,22 +27,22 @@ def list(ticker: str, export_path: Optional[str]) -> None:
     `i8 earnings list --ticker AAPL`
 
     """
-    console = Console()
-    with console.status("Fetching data...", spinner="material"):
-        df = get_hist_earnings_df(ticker, size=10)
+    earnings_list: EarningsListResult = earnings_api.list(ticker, 10)
     if export_path:
         if export_path.split(".")[-1] == "html":
-            df_formatted = format_hist_earnings_df(df, "console")
-            table = df2Table(df_formatted)
+            df = earnings_list.to_df()
+            table = df2Table(df)
             export_to_html(table, export_path)
-            return
-        export_data(
-            format_hist_earnings_df(df, "store"),
-            export_path,
-            column_width=18,
-            column_format=APP_SETTINGS["styles"]["xlsx"]["financials"]["column"],
-        )
-        return
-    df_formatted = format_hist_earnings_df(df, "console")
-    table = df2Table(df_formatted)
-    console.print(table)
+        else:
+            export_data(
+                earnings_list.to_df("raw"),
+                export_path,
+                column_width=18,
+                column_format=APP_SETTINGS["styles"]["xlsx"]["financials"]["column"],
+            )
+    else:
+        df = earnings_list.to_df()
+        earnings_list.to_console()
```

Comment:
This should already print the table in console and the following lines should not be necessary. Also, make sure you use "humanize" format.